flushers/kinesis.js

```diff
@@ -45,30 +41,21 @@ class KinesisFlusher {
   call (data, callback) {
     const kinesisMessages = data.batch.map((record) => {
       var pk = (1.0 * Math.random()).toString().replace('.', '')
-      var ehk = (1.0 * Math.random()).toString().replace('.', '')
-
-      while (ehk[0] === '0' && ehk.length > 0) {
-        ehk = ehk.substring(1)
-      }
 
       return {
-        'PartitionKey': pk,
-        'ExplicitHashKey': ehk,
-        'Data': JSON.stringify(record)
+        partitionKey: pk,
+        data: JSON.stringify(record)
       }
     })
 
     // The callback is envoked when the number of records supplied
     // exceeds the Kinesis maximum record size
-    aggregator.aggregateRecords(
+    aggregate(
       kinesisMessages,
-      this.sendMessageToKinesis.bind(this, callback),
+      this.sendMessageToKinesis.bind(this),
       noop,
-      noop
+      callback
     )
-
-    // flush any final messages that were under the emission threshold
-    aggregator.flushBufferedRecords(this.sendMessageToKinesis.bind(this, callback))
```

Comment:
We don't need to force flush here anymore, the `aggregate` function will take care of that.